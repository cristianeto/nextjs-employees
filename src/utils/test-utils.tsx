import React, { FC, ReactElement } from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { SWRConfig } from 'swr';
import { makeStore } from '../redux/store';
import { IWrapperProvider } from '@interfaces';

const store = makeStore();

// The rest of provider must be here, For example: SessionProvider of next/auth
const allProviders: FC<IWrapperProvider> = ({ children }) => {
  return (
    <SWRConfig value={{ dedupingInterval: 0 }}>
      <Provider store={store}>{children}</Provider>
    </SWRConfig>
  );
};
// using allProviders
const renderWithStore = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: allProviders, ...options });
// normal render
const Providers: FC<IWrapperProvider> = ({ children }) => children;
const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { customRender as render, renderWithStore };
