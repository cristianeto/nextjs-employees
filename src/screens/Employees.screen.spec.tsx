import { RouterContext } from 'next/dist/shared/lib/router-context';
import EmployeesScreen from './Employees.screen';
import { createMockRouter } from '@mocks';
import { renderWithStore, waitFor } from '@utils/test-utils';

describe('<EmployeesScreen/>', () => {
  it('should render the EmployeesScreen component', () => {
    const { baseElement, getByRole, findByTestId } = renderWithStore(
      <RouterContext.Provider value={createMockRouter({})}>
        <EmployeesScreen />
      </RouterContext.Provider>,
    );
    waitFor(async () => {
      const buttonAdd = getByRole('button', { name: /new employee/i });
      expect(buttonAdd).toBeInTheDocument();
      const list = await findByTestId('list-employee');
      const form = await findByTestId('employee-form');
      expect(list).toBeInTheDocument();
      expect(form).toBeInTheDocument();
    });
    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
