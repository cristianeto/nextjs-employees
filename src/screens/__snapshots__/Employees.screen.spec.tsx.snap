// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployeesScreen/> should render the EmployeesScreen component 1`] = `
<body>
  <div>
    <nav>
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/auth/login"
          >
            Login
          </a>
        </li>
      </ul>
    </nav>
    <h1
      class="chakra-heading css-0"
    >
      Employees Page
    </h1>
    <button
      class="chakra-button css-0"
      type="button"
    >
      New employee
    </button>
    <div
      class="chakra-table__container css-0"
    >
      <table
        class="chakra-table css-0"
        role="table"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <th
              class="css-0"
            >
              #
            </th>
            <th
              class="css-0"
            >
              Name
            </th>
            <th
              class="css-0"
            >
              Lastname
            </th>
            <th
              class="css-0"
            >
              Email
            </th>
            <th
              class="css-0"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody
          class="css-0"
        />
        <tfoot
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <th
              class="css-0"
            >
              #
            </th>
            <th
              class="css-0"
            >
              Name
            </th>
            <th
              class="css-0"
            >
              Lastname
            </th>
            <th
              class="css-0"
            >
              Email
            </th>
            <th
              class="css-0"
            >
              Actions
            </th>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</body>
`;
